base/loading.jl

```diff
@@ -865,8 +865,26 @@ For more details regarding code loading, see the manual.
 function require(into::Module, mod::Symbol)
     uuidkey = identify_package(into, String(mod))
     # Core.println("require($(PkgId(into)), $mod) -> $uuidkey")
-    uuidkey === nothing &&
-        throw(ArgumentError("Module $mod not found in current path.\nRun `Pkg.add(\"$mod\")` to install the $mod package."))
+    if uuidkey === nothing
+        where = PkgId(into)
+        if where.uuid === nothing
+            throw(ArgumentError("""
+                Package $mod not found in current path:
+                 - Activate the `pkg>` prompt by pressing the `]` key.
+                 - Run `pkg> add $mod` to install the $mod package.
+                """))
+        else
+            throw(ArgumentError("""
+                Package $(where.name) does not have $mod in its dependencies:
+                 - Activate the `pkg>` prompt by pressing the `]` key.
+                 - If you have $(where.name) checked out for development and have
+                   added $mod as a dependency but haven't updated your primary
+                   environment's manifest file, try doing `pkg> up --fixed` to
```

Comment:
Seems there isn't a `Pkg.update(fixed=true)` or similar, but that would make a good addition that would simplify this as well.

Reply:
Yeah the API for setting upgrade on levels is currently not so user friendly

Perhaps there should just be a`resolve` that does this.

Also, arguably the message here should have the `--manifest` flag.